docs/csharp/nullable-references.md

```diff
@@ -55,25 +96,26 @@ The compiler uses the following rules in a disabled nullable context:
 
 The behavior is the same as previous versions of C#.
 
-The compiler uses the following rules in an enabled nullable context:
+The compiler uses the following rules in an enabled nullable annotation context:
 
 - Any variable of a reference type is a **non-nullable reference**.
 - Any non-nullable reference may be dereferenced safely.
 - Any nullable reference type (noted by `?` after the type in the variable declaration) may be null. Static analysis determines if the value is known to be non-null when it is dereferenced. If not, the compiler warns you.
 - You can use the null-forgiving operator to declare that a nullable reference isn't null.
 
-A richer grammar is proposed for nullable contexts and will be available in future previews. For more information about nullable contexts, see the [nullable reference specification](https://github.com/dotnet/csharplang/blob/master/proposals/nullable-reference-types-specification.md#nullable-contexts) draft.
+In an enabled nullable annotation context, the `?` character appended to a reference type declares a **nullable reference type**. The **null forgiveness operator** (`!`) may be appended to an expression to declare that the expression is not null.
 
-## Null states of nullable references
+## Nullable warning context
 
-The compiler uses static analysis to determine the **null state** of any nullable reference. The null state is either **not null** or **maybe null**. If you dereference a nullable reference when the compiler has determined it's **maybe null**, the compiler warns you. The state of a nullable reference is **maybe null** unless the compiler can determine one of two conditions:
+The nullable warning context is distinct from the nullable annotation context. Warnings can be enabled even when the new annotations are disabled. The compiler uses static flow analysis to determine the **null state** of any reference. The null state is either **not null** or **maybe null** when the *nullable warning context* is not **disabled**. If you dereference a reference when the compiler has determined it's **maybe null**, the compiler warns you. The state of a reference is **maybe null** unless the compiler can determine one of two conditions:
 
 1. The variable has been definitely assigned to a non-null value.
 1. The variable has been checked against null before de-referencing it.
```

Comment:
It may be worth mentioning expressions instead of variables. The fact that `if (foo.Address != null)` changes whether I can use `foo.Address.Country` later without a warning was one of the aspects of nullability that most surprised me the first time I saw it.